backend/app/controllers/tasks.js

```diff
@@ -350,3 +375,73 @@ function* checkTaskData(req) {
         yield * common.checkDuplicateTask(req, req.body.stepId, req.body.uoaId, req.body.productId);
     }
 }
+
+function* updateCurrentStepId(req, insertedTaskId) {
+    var thunkQuery = req.thunkQuery;
+
+    // var essenceId = yield * getEssenceId(req, 'Tasks');
+    var product = yield * common.getEntity(req, req.body.productId, Product, 'id');
+
+    //TODO: Get survey from survery service if needed
+    // var survey = yield * common.getEntity(req, product.surveyId, Survey, 'id');
+    if (product.status !== 2) { // not suspended
+        yield thunkQuery(
+            ProductUOA.update({
+                isComplete: false
+            }).where(ProductUOA.productId.equals(req.body.productId))
+        );
+    }
+
+    var result;
+    var minStepPositionQuery = WorkflowStep
+        .select(
+            WorkflowStep.position,
+            Task.stepId,
+            Task.id
+        )
+        .from(WorkflowStep
+            .join(Task).on(Task.stepId.equals(WorkflowStep.id))
+        )
+        .where(Task.productId.equals(req.body.productId))
+        .and(Task.uoaId.equals(req.body.uoaId))
+        .order(WorkflowStep.position);
+    result = yield thunkQuery(minStepPositionQuery);
+
+    if (!_.first(result)) {
+        debug('Not found min step position for productId `' + req.body.productId + '`');
+        return null;
```

Comment:
Do we want to return null here or just throw an error? 
`throw new HttpError(403, 'Not found min step position for productId `' + req.body.productId + '`')`

Reply:
Sure.

Reply:
Done.